desc/objectives/_geometry.py

```diff
@@ -1077,3 +1078,224 @@ def compute(self, *args, **kwargs):
+        )
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        self._surface_coords = self._surface.compute("x", basis="xyz")["x"]
+        self._profiles = get_profiles(
+            self._data_keys,
+            eq=eq,
+            grid=plasma_grid,
+            has_axis=plasma_grid.axis.size or surface_grid.axis.size,
+            obj=eq,
+        )
+        self._transforms = get_transforms(
+            self._data_keys,
+            eq=eq,
+            grid=plasma_grid,
+            has_axis=plasma_grid.axis.size or surface_grid.axis.size,
+            obj=eq,
+        )
+
+        self._constants = {
+            "transforms": self._transforms,
+            "profiles": self._profiles,
+            "surface_coords": self._surface_coords,
+        }
+
+        timer.stop("Precomputing transforms")
+        if verbose > 1:
+            timer.disp("Precomputing transforms")
+
+        if self._normalize:
+            scales = compute_scaling_factors(eq)
+            self._normalization = (scales["B"] * scales["a"]) ** 2
+
+        super().build(eq=eq, use_jit=use_jit, verbose=verbose)
+
+    def compute(self, *args, **kwargs):
+        """Computes the flux-surface variance of |B|*d_min.
+
+        Parameters
+        ----------
+        R_lmn : ndarray
+            Spectral coefficients of R(rho,theta,zeta) -- flux surface R coordinate (m).
+        Z_lmn : ndarray
+            Spectral coefficients of Z(rho,theta,zeta) -- flux surface Z coordinate (m).
+        L_lmn : ndarray
+            Spectral coefficients of lambda(rho,theta,zeta) -- poloidal stream function.
+        i_l : ndarray
+            Spectral coefficients of iota(rho) -- rotational transform profile.
+        c_l : ndarray
+            Spectral coefficients of I(rho) -- toroidal current profile.
+        Psi : float
+            Total toroidal magnetic flux within the last closed flux surface (Wb).
+
+        Returns
+        -------
+        f : ndarray
+            value of |B|*d_min at each point on the plasma .
+
+        """
+        params, constants = self._parse_args(*args, **kwargs)
+        if constants is None:
+            constants = self.constants
+
+        data = compute_fun(
+            "desc.equilibrium.equilibrium.Equilibrium",
+            self._data_keys,
+            params=params,
+            transforms=self._transforms,
+            profiles=self._profiles,
+        )
+
+        d = jnp.linalg.norm(
+            jnp.array([data["X"], data["Y"], data["Z"]]).T[:, None, :]
+            - constants["surface_coords"][None, :, :],
+            axis=-1,
+        )
+
+        if self._use_softmin:  # do softmin
+            dmin_data = jnp.apply_along_axis(softmin, -1, d, self._alpha)
+        else:  # do hardmin
+            dmin_data = d.min(axis=-1)
+
+        B_dmin_data = jnp.array([d_min * B for d_min, B in zip(dmin_data, data["|B|"])])
```

Comment:
can't these just be broadcasted normally?

Reply:
yes, though when I make that change our test fails... it must not be a good test then

Reply:
It might also be something weird with array shaping. I'd double check the sizes and values of things to make sure it's doing what you want.